pipeline/processor/postprocessor.py

```diff
@@ -0,0 +1,663 @@
+            regex_match = re.search(r"\w+(-|–){1}$", l)
+
+            if current:
+                word_fragment = current.rstrip("-")
+                # TODO: Handle non-English words
+                if word_fragment in english_words_set:
+                    # Check if the word with hyphen removed is an english
+                    # word and if it is, make this the first word of the newline
+                    # without the hyphen (e.g. repair-ing)
+                    newline_first_word = word_fragment + l.split(" ")[0].lstrip("-")
+                    if newline_first_word in english_words_set:
+                        new_list[ix] = newline_first_word + l.split(" ")[1:]
+                    # Otherwise, keep the hyphenation but put it on a newline e.g. post-processing.
+                    else:
+                        new_list[ix] = word_fragment + "-" + l
+                else:
+                    new_list[ix] = word_fragment + l
+                # Reset.
+                current = None
+            if regex_match:
+                # Strip matching regex from the end of the string.
+                new_list[ix] = l[: regex_match.start()]
+                current = regex_match[0]
+        return new_list
+
+    def process(self, contents: Document) -> Document:
+        contents = contents.to_dict()
+        for ix, page in enumerate(contents["pages"]):
+            for text_block in page["text_blocks"]:
+                text_block["text"] = self._rewrap_hyphenated_words(text_block["text"])
+        contents = Document.from_dict(contents)
+        return contents
+
+
+class AdobeTextStylingPostProcessor(PostProcessor):
+    """
+    Some semantic passages have separate blocks for styling markers such
+    as underlines, superscripts and subscripts. We want to group such cases
+    into single contiguous blocks. However, we want to keep the styling information
+    because it's often relevant to semantics. For example, CO2 is often represented
+    using a subscript. For now, we keep everything and indicate the styling with
+    HTML style tags inline.
+
+    """
+
+    def __init__(self):
+        super(AdobeTextStylingPostProcessor, self).__init__()
+
+    @staticmethod
+    def _classify_text_block_styling(text_block: TextBlock) -> Optional[str]:
+        """
+        Get text block styling, if present.
+
+        Args:
+            text_block:
+
+        Returns:
+
+        """
+        if not text_block["custom_attributes"]:
+            return None
+
+        if text_block["custom_attributes"].get("BaselineShift", 0) < 0:
+            return "subscript"
+        elif text_block["custom_attributes"].get("TextDecorationType") == "Underline":
+            return "underline"
+        elif text_block["custom_attributes"].get("TextPosition") == "Sup":
+            return "superscript"
+        else:
+            return None
+
+    @staticmethod
+    def _add_text_styling_markers(text: str, styling: str) -> str:
```

Comment:
It looks like this method can be deleted as we're using the spacy-like styling objects now?